### PR TITLE
feat: upgrade to beta-4 (fuel-core 0.20.4)

### LINKED ISSUE #826: Update fuel-wallet to support fuel-core v0.20

### PR DESCRIPTION
**Pull request by LuizAsFight**
Closes #826 

Closes FRO-258
Closes FRO-449

### CODE REVIEW

File: packages/app/src/systems/Core/__tests__/utils/error.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export function byPassDataSpaceError() {
+  // TODO: remove completely this after fuel-ui `dataSpace` problem gets fixed
+  beforeEach(() => {
+    const consoleError = console.error;
```
Comment (luizstacio) [inline packages/app/src/systems/Core/__tests__/utils/error.ts:4]: Add a comment here to remove the warning

File: packages/app/src/systems/Send/components/Send/Send.test.tsx [hunk 2]
```diff
@@ -31,7 +31,6 @@ describe('Send', () => {
     it('should show transaction details by default', async () => {
       render(<Select />, { wrapper: TestWrapper });
       await waitFor(() => findAssetSelect());
-      expect(await screen.findByText('Fee (network)')).toBeInTheDocument();
```
Comment (luizstacio) [inline packages/app/src/systems/Send/components/Send/Send.test.tsx:34]: Should we remove this?

**Review by luizstacio — CHANGES_REQUESTED** [review]
We just need to update fuels to a released version

File: packages/app/src/systems/Send/components/Send/Send.test.tsx (hunk 2 above)
Comment (LuizAsFight) [reply packages/app/src/systems/Send/components/Send/Send.test.tsx:34]: maybe we should remove this whole test as we have it in e2e already. fixed it for now

**Review by luizstacio — APPROVED** [review]

**Pull request merged**